File: c/PRAK604-2310817310004-MSAMILRENDYNORSALEH.c
#include <stdio.h>
#include <string.h>

void decryptMessage(char *kode, char *pesan) {
    int panjangkode = strlen(kode);
    int panjangpesan = strlen(pesan);

    if (panjangkode != panjangpesan) {
        printf("Panjang kalimat berbeda, pesan palsu\n");
        return;
    }
    char hasildeskripsi[panjangpesan + 1];
    int bintang = 0, hashtag = 0;

    for (int i = 0; i < panjangpesan; i++) {
        if (kode[i] == pesan[i]) {
            if (kode[i] == ' ') {
                hasildeskripsi[i] = ' ';
            } else {
                hasildeskripsi[i] = '*';
                bintang++;
            }
        } else {
            hasildeskripsi[i] = '#';
            hashtag++;
        }
    }
    hasildeskripsi[panjangpesan] = '\0';

    printf("%s\n", hasildeskripsi);
    printf("* = %d\n", bintang);
    printf("# = %d\n", hashtag);

    if (bintang >= hashtag) {
        printf("Pesan Asli\n");
    } else {
        printf("Pesan Palsu\n");
    }
}

int main() {
    char kode[100], pesan[100];

    printf("");
    fgets(kode, sizeof(kode), stdin);
    kode[strcspn(kode, "\n")] = '\0';

    printf("");
    fgets(pesan, sizeof(pesan), stdin);
    pesan[strcspn(pesan, "\n")] = '\0';  

    decryptMessage(kode, pesan);
    return 0;
}

File: python/PRAK604-2310817310004-MSAMILRENDYNORSALEH.py
def decrypt_message(kode, pesan):
    panjangkode = len(kode)
    panjangpesan = len(pesan)

    if panjangkode != panjangpesan:
        print("Panjang kalimat berbeda, pesan palsu")
        return

    hasildeskripsi = [""] * panjangpesan
    bintang = 0
    hashtag = 0

    for i in range(panjangpesan):
        if kode[i] == pesan[i]:
            if kode[i] == ' ':
                hasildeskripsi[i] = ' '
            else:
                hasildeskripsi[i] = '*'
                bintang += 1
        else:
            hasildeskripsi[i] = '#'
            hashtag += 1

    hasildeskripsi = "".join(hasildeskripsi)

    print(hasildeskripsi)
    print(f"* = {bintang}")
    print(f"# = {hashtag}")

    if bintang >= hashtag:
        print("Pesan Asli")
    else:
        print("Pesan Palsu")

def main():
    kode = input("")
    pesan = input("")

    decrypt_message(kode, pesan)

if __name__ == "__main__":
    main()
